crates/transport_tx5/src/test.rs

```diff
@@ -0,0 +1,407 @@
+use super::*;
+use std::sync::Mutex;
+
+// We don't need or want to test all of tx5 in here... that should be done
+// in the tx5 repo. Here we should only test the kitsune2 integration of tx5.
+//
+// Specifically:
+//
+// - That new_listening_address is called if the sbd server is restarted
+// - That peer connect / disconnect are invoked appropriately.
+// - That messages can be send / received.
+// - That preflight generation and checking work, which are a little weird
+//   because in kitsune2 the check logic is handled in the same recv_data
+//   callback, where tx5 handles it as a special callback.
+
+struct Test {
+    pub srv: Option<sbd_server::SbdServer>,
+    pub port: u16,
+    pub builder: Arc<kitsune2_api::builder::Builder>,
+}
+
+impl Test {
+    pub async fn new() -> Self {
+        let mut this = Self {
+            srv: None,
+            port: 0,
+            builder: Arc::new(kitsune2_core::default_test_builder()),
+        };
+
+        this.restart().await;
+
+        let builder = kitsune2_api::builder::Builder {
+            transport: Tx5TransportFactory::create(),
+            ..kitsune2_core::default_test_builder()
+        };
+
+        builder
+            .config
+            .set_module_config(&config::Tx5TransportModConfig {
+                tx5_transport: config::Tx5TransportConfig {
+                    server_url: format!("ws://127.0.0.1:{}", this.port),
+                },
+            })
+            .unwrap();
+
+        this.builder = Arc::new(builder);
+
+        this
+    }
+
+    pub async fn restart(&mut self) {
```

Comment:
On first start this will use port `0` and on subsequent restarts we need to re-use the port that we assigned the first time so that the transport config doesn't need updating for peers to reconnect?

A quick doc comment about what this is doing might be useful. If something does happen to claim the port while we're restarting (not that likely) then we could see the occasional flake from this and it'd be nice to have a doc comment to remind us what it does.

Reply:
Done: https://github.com/holochain/kitsune2/pull/75/commits/3fcf51a5a7a02739badd0e83a54021941b7bd412